aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -286,7 +287,7 @@ void nll_loss_forward_out_cuda_template(
 
   auto weight_ = weight.defined() ? weight.contiguous() : weight;
 
-  if (reduction == Reduction::None & n_dims == 2) {
+  if (reduction == Reduction::None && n_dims == 2) {
```

Comment:
here too :0 Richard says we should have a linter rule for catching this sort of thing- seems like a good idea

Reply:
The compiler does warn about these I think. Maybe we want to make these warnings errors.

Reply:
We should probably compile pytorch with -Werror for good practice. The downside is pytorch already emits a lot of warnings so fixing pytorch to get it into a good shape for this will take a while